src/main/java/org/broadinstitute/hellbender/utils/clipping/ClippingOp.java

```diff
@@ -176,34 +177,46 @@ private GATKRead applyREVERT_SOFTCLIPPED_BASES(final GATKRead read) {
     }
 
     /**
-     * Given a cigar string, get the number of bases hard or soft clipped at the start
+     * Given two cigar strings corresponding to the a read from a read before and after soft-clipping, returns an integer
+     * corresponding to the number of reference bases that the new string must be offset by in order to have the
+     * correct start according to the reference.
+     *
+     * @param clippedCigar the new cigar string after clipping
+     * @param oldCigar the cigar string of the read before it was soft clipped
      */
-    private int getNewAlignmentStartOffset(final Cigar __cigar, final Cigar __oldCigar) {
-        int num = 0;
-        for (final CigarElement e : __cigar.getCigarElements()) {
+    @VisibleForTesting
+    static int getNewAlignmentStartOffset(final Cigar clippedCigar, final Cigar oldCigar) {
+        int readBasesBeforeReference = 0; // The number of reference bases consumed on the old cigar before read bases are consumed
```

Comment:
Hmm, is this name accurate? The description says "The number of reference bases consumed on the old cigar before read bases are consumed", but the name implies the opposite (read bases before reference bases are consumed). Which is it?